backend/src/db/repositories/complianceRepository.ts

```diff
@@ -82,6 +82,29 @@ const mongoComplianceRepository: ComplianceDbRepository = {
     }
   },
 
+  async editOrSubmitDraftForm(draftId: string, updatedData: Partial<ComplianceReport>): Promise<void> {
+    try {
+
+      const draft = await Compliance.findOne({ uniqueId: draftId });
+
+      if (!draft) {
+        throw new Error(`Draft with unique ID ${draftId} does not exist.`);
+      }
+      if (draft.expirationDate && draft.expirationDate < new Date()) {
+        throw new Error("You cannot edit an expired draft form.");
+      }
+      if (draft.status !== StatusEnum.DRAFT) {
+        throw new Error("You cannot edit a form that is not in the draft status.");
+      }
+
+      await Compliance.updateOne({ uniqueId: draftId }, updatedData);
+
+    } catch (error) {
+      console.error(`Error updating the draft form with unique ID ${draftId}:`, error);
+      throw error;
+    }
+  },
```

Comment:
I don't thin I get this function. the name says `editOrSubmitDraftForm` but looking at the logic it only update the draft and fail if it's not already in database. I also don't see anything specific for draft submission. Assuming that draft submission is  basically meaning that status of the form is changed from DRAFT to REVIEW (or other relevant status), I believe that it might be confusing later on in application lifecycle (e.g. missing fields in DRAFT validation). I think these functionalities should be distinct in the system. Let me know if you'd like to dicsuss it further.

Reply:
Yes, it is something I should have change. At the beginning of the work, we thought that you should be able to submit the form without creating a draft. Since it is not the case we have to change the name to `editDraftForm`.